internal/libyaml/api_test.go

```diff
@@ -0,0 +1,604 @@
+			val := parseBoolArg(t, arg)
+			callArgs = append(callArgs, reflect.ValueOf(val))
+		} else if paramType.Kind() == reflect.Slice && paramType.Elem().Kind() == reflect.Uint8 {
+			// Byte slice parameter
+			str := parseStringArg(t, arg, useBytes)
+			callArgs = append(callArgs, reflect.ValueOf([]byte(str)))
+		} else {
+			// Try parsing as constant/int
+			val := parseArg(t, arg)
+			convertedVal := reflect.ValueOf(val).Convert(paramType)
+			callArgs = append(callArgs, convertedVal)
+		}
+	}
+
+	// Call the method
+	m.Call(callArgs)
+}
+
+// parseArg parses an argument which could be int, bool, or string constant
+func parseArg(t *testing.T, arg interface{}) int {
+	t.Helper()
+	switch v := arg.(type) {
+	case int:
+		return v
+	case string:
+		if looksLikeConstant(v) {
+			return parseConstant(t, v)
+		}
+		// Try parsing as int
+		if val, err := strconv.Atoi(v); err == nil {
+			return val
+		}
+		t.Fatalf("cannot parse arg as int: %v", arg)
+	default:
+		t.Fatalf("unsupported arg type: %T", arg)
+	}
+	return 0
+}
+
+// parseBoolArg parses a boolean argument
+func parseBoolArg(t *testing.T, arg interface{}) bool {
+	t.Helper()
+	switch v := arg.(type) {
+	case bool:
+		return v
+	case string:
+		switch v {
+		case "true":
+			return true
+		case "false":
+			return false
+		default:
+			t.Fatalf("cannot parse string as bool (expected 'true' or 'false'): %q", v)
+		}
+	default:
+		t.Fatalf("cannot parse arg as bool: %v (type %T)", arg, arg)
+	}
+	return false
+}
+
+// parseStringArg parses a string argument
+func parseStringArg(t *testing.T, arg interface{}, useBytes bool) string {
+	t.Helper()
+	switch v := arg.(type) {
+	case string:
+		return v
+	default:
+		t.Fatalf("cannot parse arg as string: %v (type %T)", arg, arg)
+	}
+	return ""
+}
+
+// looksLikeConstant checks if a string looks like a constant name
+func looksLikeConstant(s string) bool {
+	if len(s) == 0 {
+		return false
+	}
+	// Check if it's all uppercase letters, digits, and underscores
+	for _, c := range s {
+		if !((c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || c == '_') {
+			return false
+		}
+	}
+	return true
+}
+
+// parseConstant parses a constant name to its integer value
+func parseConstant(t *testing.T, name string) int {
+	t.Helper()
+	// Handle boolean values
+	if name == "true" {
+		return 1
+	}
+	if name == "false" {
+		return 0
+	}
+
+	// Try parsing as int first
+	if val, err := strconv.Atoi(name); err == nil {
+		return val
+	}
+
+	// Use IntOrStr to parse other constants
+	ios := IntOrStr{}
+	err := ios.FromValue(name)
+	if err != nil {
+		t.Fatalf("failed to parse constant %q: %v", name, err)
+	}
+	return ios.Value
+}
+
+// runFieldChecks runs field checks on an object
+func runFieldChecks(t *testing.T, obj interface{}, checks []FieldCheck) {
+	t.Helper()
+
+	v := reflect.ValueOf(obj)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	for _, check := range checks {
+		// Handle nil checks
+		if len(check.Nil) >= 2 {
```

Comment:
Nitpicking here: shouldn't we report as error with a t.Fatal any field of the struct that require to have a len() > 0 but len() != 2 ?

Something like this (pseudo code)

```go
func hasLength(t *testing.T, s int, v []any) bool {

  if len(v) == 0 {
    return false
  }

  if len(v) != s {
    return true
  }

  t.Fatalf("length differs: want %d got %d")
}
```

Then this can be applied on all if statement

```suggestion
		if hasLength(t, check.Nil, 2) {
```

Reply:
Of simply checking

```go
if len(check.Nil) >= 1 {
   if len(check.Nil) != 2 {
      t.Fatalf("Nil check expects exactly 2 args, got %d", len(check.Nil))
   }
   
   // the current code
}
```

Reply:
d2003eb